### ISSUE TITLE
how to update the SVGO to latest version? 

### ISSUE THREAD

**Issue by Manitej66** [rtivital/svgo-browser#16]
Thanks for the amazing repo! we need this package in our app, i'm just worried if the svgo used here is too old?

P.S: we use mantine at work! kudos!

**Comment by rtivital** [comment]
I believe you can use svgo in the browser without this package, svgo added this feature a while back

**Comment by Manitej66** [comment]
Thanks for the prompt reply! 

I did try using the bundled browser version of the svgo package but for some reason it’s not working with nextjs. I’m getting random page load errors. Probably due to SWC maybe.

**Comment by Manitej66** [comment]
Tried importing like below: 
```
import { optimize } from 'svgo/dist/svgo.browser.js'
```

it haven’t thrown any errors but when using the optimize function, my app was getting stuck and not working at all.

**Comment by rtivital** [comment]
Well, then you can try your luck with the svgo-browser then – there are not much difference between the svgo 1 and 3 anyway.

**Comment by AurelianSpodarec** [comment]
@Manitej66 Have you found a solution? I'm also trying to use SVGO with NextJS, and no luck. Can't even get this to work, yet.

**Comment by Manitej66** [comment]
@AurelianSpodarec I got it working this way (using `svgo` v3):

You need to import it like below:
```js
import { optimize } from "svgo/dist/svgo.browser"
```

in `next.config.mjs`:
```js
transpilePackages: ["svgo"]
```

`svgo.d.ts`:
```js
declare module "svgo/dist/svgo.browser" {
  import type svgo from "svgo";
  export const optimize: typeof svgo.optimize;
}
```

**Comment by AurelianSpodarec** [comment]
Hey @Manitej66! Thank you very much!

I did something else to make it work, but this is cearly the better solution :) 

Seems like an easy fix, but who knows what transpilePackages is xD

Front-end dev who doesn't know webpack hehe. Appreciate the reply!

**Comment by AurelianSpodarec** [comment]
Got last question, @Manitej66 Have you  used SVGR perhaps? :D I want to compile the optimized SVG to then React/JSX, data URI code with SVGR - but that too, has issues.

I get this with SVGR, even if I do this:   transpilePackages: ["svgo", "@svgr/core"]

I suppose this is something different, I wonder if its possible to make this work as well in the browser, without making API calls.

Failed to compile
```

./node_modules/cosmiconfig/dist/Explorer.js:7:1
Module not found: Can't resolve 'fs/promises'
   5 | Object.defineProperty(exports, "__esModule", { value: true });
   6 | exports.Explorer = void 0;
>  7 | const promises_1 = __importDefault(require("fs/promises"));
     | ^
   8 | const path_1 = __importDefault(require("path"));
   9 | const path_type_1 = require("path-type");
  10 | const ExplorerBase_js_1 = require("./ExplorerBase.js");

https://nextjs.org/docs/messages/module-not-found
```

**Comment by Manitej66** [comment]
Yes I did used SVGR. I spent several days to get the unoptimised SVG to optimised JSX conversion working in the browser. To use SVGR, you need to use version 5. If not, you'll get the errors you are getting. 

Install `@svgr/core` and `@svgr/plugin-jsx` version `5.5.0`

Usage:

```js
import convert from "@svgr/core/lib/convert";
import JSXPlugin from "@svgr/plugin-jsx";

const jsx = await new Promise<string>((resolve) => {
    convert(svg, {
      prettier: false,
      svgo: false,
      jsxRuntime: "automatic",
      plugins: [JSXPlugin],
    }).then((jsx: string) => {
       console.log(jsx);
     })});
```

You might need to fix the TS lint errors :)

**Comment by AurelianSpodarec** [comment]
@Manitej66 Hmmmm. Not sure if I'm doing it right.

I got versions
```js
    "@svgr/core": "5.5.0",
    "@svgr/plugin-jsx": "5.5.0",
```

And the 'convert' file is in the NPM - unlike in other versions.

Then when I import it (I don't even  include the other one or the code, just a simple import of the below)

```js
import convert from "@svgr/core/lib/convert";
```

I get this:

```js
Failed to compile

./node_modules/cosmiconfig/dist/readFile.js:9:1
Module not found: Can't resolve 'fs'
   7 | exports.readFileSync = readFileSync;
   8 |
>  9 | var _fs = _interopRequireDefault(require("fs"));
     | ^
  10 |
  11 | function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
  12 |

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/cosmiconfig/dist/Explorer.js
./node_modules/cosmiconfig/dist/index.js
./node_modules/@svgr/core/lib/config.js
./node_modules/@svgr/core/lib/convert.js
```

Config:
```js
import type { NextConfig } from "next"
import { webpack } from "next/dist/compiled/webpack/webpack";

const nextConfig: NextConfig = {
  transpilePackages: ["svgo"],
};

export default nextConfig
```

Should I display this in the config, the FS? And then other things etc...?

**Comment by Manitej66** [comment]
Ah yes, I forgot to include the additional config that you need to add. As you only want the SVGR to work in browser we need to explicitly tell webpack to not bundle the `fs` and `module` modules:

add the following in `next.config.mjs`:

```js
  webpack: (config, { isServer }) => {
    if (!isServer) {
      config.resolve = {
        ...config.resolve,
        fallback: {
          ...config.resolve.fallback,
          fs: false,
          module: false,
        },
      };
    }

    return config;
  },
 ```

**Comment by AurelianSpodarec** [comment]
@Manitej66 Hmm :D I wonder, is there anything else you got?

I get this error in comand line

```js
./node_modules/@babel/core/lib/config/files/module-types.js
require.extensions is not supported by webpack. Use a loader instead.
```

and this in browser:

```js
Unhandled Runtime Error

TypeError: Cannot assign to read only property '__esModule' of object '#<Object>'

./node_modules/@svgr/babel-plugin-transform-svg-component/lib/index.js
```

What Version of NextJS are you using? Was it from the Vercel original template?

I've also cleared cache, maybe I need to remove node modules and re-install - will try that now. (nope, didn't work)

**Comment by Manitej66** [comment]
@AurelianSpodarec hmm, never got this error before. 
I'm using next V14. Also worked with V13. Haven't tested with V15. 

But it seems its related to webpack

**Comment by AurelianSpodarec** [comment]
@Manitej66 Damn, I forgot to remove the transpile stuff. I put the svgr into transpile as well.

So I've managed to compile it on the front end.

Though still go this notification:

```
Import trace for requested module:
./node_modules/@babel/core/lib/config/files/module-types.js
./node_modules/@babel/core/lib/config/files/configuration.js
./node_modules/@babel/core/lib/config/files/index.js
./node_modules/@babel/core/lib/index.js
./node_modules/@svgr/plugin-jsx/lib/index.js
./node_modules/@svgr/core/lib/plugins.js
./node_modules/@svgr/core/lib/convert.js
./src/app/page/_components/PageContent.tsx

./node_modules/@babel/core/lib/config/files/module-types.js
require.extensions is not supported by webpack. Use a loader instead.
```

Guess gonna have to figure that out myself in that case.

I seriously appreciate your help!

Been on this for a few days myself, first doing it manually, then looking for some libs that do it for me - it seems like these libs can work just fine in the browsers, but they were never intended to use like that it seems, if the maintainers wanted to, they could probably add the functionality to it I'm sure.

But yeah, really appriciate your help with this. Probably woudln't have figured that out myself, no idea how you figured out that you can use the convert function from v 5.5

**Comment by Manitej66** [comment]
@AurelianSpodarec 🙏 glad it worked. Yeah, you should not add svgr in transpile array. 

I spent so many days trying to get it working in the browser with latest version. But then I found https://transform.tools/ this app has SVG to JSX conversion and it doesn't do any network reqs. So I digged into the codebase and tried using same versions as theirs and it worked.

**Comment by AurelianSpodarec** [comment]
@Manitej66 Nice! I've done the same for other apps, and I saw they are using SVGO but wasn't too sure about the SVGR.

Interesting though - you've done better digging than I have! 

I think this is probably the best right - there are some standalone packages but... probably not as good I'd assume.

What are you building anyway? :D

**Comment by Manitej66** [comment]
@AurelianSpodarec at work I had the requirement to display SVG in a PDF. So I did this logic that converts RAW svg to jsx that can be displayed using react-pdf.

**Comment by AurelianSpodarec** [comment]
Nice! Interesting. Problem solving power! :D
